examples/nas/textnas/dataloader.py

```diff
@@ -0,0 +1,334 @@
+                    pos = i
+                    for j in range(i, 0, -1):
+                        if text[j] == ' ':
+                            pos = j
+                            break
+                    self.word = text[pos + 1:i]
+                    self.span = (left, left + 1)
+
+            # we've reached the end of the category that is the root of this subtree
+            if depth == 0 and char == ' ' and self.label == '':
+                self.label = text[pos + 1:i]
+            # we've reached the end of the scope for this bracket
+            if depth < 0:
+                break
+
+        # Fix some issues with variation in output, and one error in the treebank
+        # for a word with a punctuation POS
+        self.standardise_node()
+
+    def standardise_node(self):
+        if self.word in word_to_word_mapping:
+            self.word = word_to_word_mapping[self.word]
+
+    def __repr__(self, single_line=True, depth=0):
+        ans = ''
+        if not single_line and depth > 0:
+            ans = '\n' + depth * '\t'
+        ans += '(' + self.label
+        if self.word is not None:
+            ans += ' ' + self.word
+        for subtree in self.subtrees:
+            if single_line:
+                ans += ' '
+            ans += subtree.__repr__(single_line, depth + 1)
+        ans += ')'
+        return ans
+
+
+def read_tree(source):
+    cur_text = []
+    depth = 0
+    while True:
+        line = source.readline()
+        # Check if we are out of input
+        if line == '':
+            return None
+        # strip whitespace and only use if this contains something
+        line = line.strip()
+        if line == '':
+            continue
+        cur_text.append(line)
+        # Update depth
+        for char in line:
+            if char == '(':
+                depth += 1
+            elif char == ')':
+                depth -= 1
+        # At depth 0 we have a complete tree
+        if depth == 0:
+            tree = PTBTree()
+            tree.set_by_text(' '.join(cur_text))
+            return tree
+    return None
+
+
+def read_trees(source, max_sents=-1):
+    with open(source) as fp:
+        trees = []
+        while True:
+            tree = read_tree(fp)
+            if tree is None:
+                break
+            trees.append(tree)
+            if len(trees) >= max_sents > 0:
+                break
+        return trees
+
+
+class SSTDataset(data.Dataset):
+    def __init__(self, sents, mask, labels):
+        self.sents = sents
+        self.labels = labels
+        self.mask = mask
+
+    def __getitem__(self, index):
+        return (self.sents[index], self.mask[index]), self.labels[index]
+
+    def __len__(self):
+        return len(self.sents)
+
+
+def sst_get_id_input(content, word_id_dict, max_input_length):
+    words = content.split(" ")
+    sentence = [word_id_dict["<pad>"]] * max_input_length
+    mask = [0] * max_input_length
+    unknown = word_id_dict["unknown"]
```

Comment:
can you unify punctuation?

Reply:
There is indeed a mistake here. But I'm not sure it's related to punctuation. There are punctuations in glove.